### PR TITLE
build.plat, back.rtlil: Fix #1104 fallout.

### PR DESCRIPTION
**Pull request by wanda-phi**

### CODE REVIEW

File: amaranth/build/plat.py [hunk 1]
```diff
@@ -307,8 +309,8 @@ def toolchain_prepare(self, fragment, name, *, ports, emit_src=True, **kwargs):
         # and to incorporate the Amaranth version into generated code.
         autogenerated = f"Automatically generated by Amaranth {__version__}. Do not edit."
 
-        rtlil_text, self._name_map = rtlil.convert_fragment(fragment, ports=ports, name=name,
-                                                            emit_src=emit_src)
+        rtlil_text, self._name_map = rtlil.convert_fragment(fragment, name=name,
+                                                            emit_src=emit_src, propagate_domains=False)
```
Comment (whitequark) [inline amaranth/build/plat.py:313]: I think this might be a stale option?

Comment (whitequark) [reply amaranth/build/plat.py:313]: Oh, no, it does get passed to `prepare()` down the road doesn't it

**Review by whitequark — APPROVED** [review]
LGTM, thank you!

File: amaranth/build/plat.py (hunk 1 above)
Comment (wanda-phi) [reply amaranth/build/plat.py:313]: yup

**Pull request merged**